frontend/src/pbd-program-list/pbd-program-list.html

```diff
@@ -109,15 +118,22 @@
       },
 
       _handleCreate: function(evt) {
+        var routeData = this.$.routeData;
         var now = new Date();
         var name = 'Untitled program ' + now.toLocaleString();
-        var msg = {
-          type: 'create program',
-          program_info: {
-            name: name
-          },
+        var request = {
+          name: name,
         };
-        this.$.eventPub.publish(msg);
+        this.$.createSrv.call(request);
+      },
+
+      _handleServiceResponse(evt) {
+        var id = evt.detail.db_id;
+        window.location.replace("#/program/" + id);
```

Comment:
Could you change `window.location.replace("#/program/" + id);` with:

```js
window.history.pushState({}, null, '#/program/' + id);
window.dispatchEvent(new CustomEvent('location-changed'));
```

This was recommended in the [app-route docs](https://www.webcomponents.org/element/PolymerElements/app-route) and I tested it out on Firefox and in RWS.